### PR TITLE
Object Mother pattern #498

### PR DESCRIPTION
**Pull request by igeligel**
Added the object mother pattern and all diagrams to the context.

### CODE REVIEW

**Comment by iluwatar** [comment]
The `object-mother` module has not been added to the parent `pom.xml`.

File: object-mother/src/test/java/com/iluwatar/objectmother/test/RoyaltyObjectMotherTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  public void unsuccessfulKingFlirt() {
+    King soberUnhappyKing = RoyaltyObjectMother.createSoberUnhappyKing();
+    Queen flirtyQueen = RoyaltyObjectMother.createFlirtyQueen();
+    soberUnhappyKing.flirt(flirtyQueen);
+    assertFalse(soberUnhappyKing.isHappy());
+  }
+  
+  @Test
+  public void queenIsBlockingFlirtCauseDrunkKing() {
+    King soberUnhappyKing = RoyaltyObjectMother.createDrunkKing();
```
Comment (iluwatar) [inline object-mother/src/test/java/com/iluwatar/objectmother/test/RoyaltyObjectMotherTest.java]: Confusing naming here

File: object-mother/src/test/java/com/iluwatar/objectmother/test/RoyaltyObjectMotherTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  public void queenIsBlockingFlirt() {
+    King soberUnhappyKing = RoyaltyObjectMother.createHappyKing();
+    Queen notFlirtyQueen = RoyaltyObjectMother.createNotFlirtyQueen();
+    soberUnhappyKing.flirt(notFlirtyQueen);
+    assertFalse(soberUnhappyKing.isHappy());
+  }
+  
+  @Test
+  public void successfullKingFlirt() {
+    King soberUnhappyKing = RoyaltyObjectMother.createHappyKing();
```
Comment (iluwatar) [inline object-mother/src/test/java/com/iluwatar/objectmother/test/RoyaltyObjectMotherTest.java]: Confusing naming here

**Comment by iluwatar** [comment]
@igeligel is this ready for another review?

**Comment by igeligel** [comment]
Yeah, i have messed up the pom.xml at some commit but fixed it then. I also renamed the code parts where you wanted a change.

**Comment by iluwatar** [comment]
Looks good @igeligel. Thanks for the contribution!

**Pull request merged**